### ISSUE TITLE
Pass environmentVariables from Visual Studio kits to vcvarsall.bat

### ISSUE THREAD

**Issue by Djoulihen** [microsoft/vscode-cmake-tools#1332]
Some context: I'm trying to specify the Windows SDK version used by my visual studio kit (there is an open issue about this:  #125 ). I managed to do this with when using cmake and Ninja in Visual Studio's CMakeSettings.json by setting the WindowsSdkVersion environment variable like this:
```
	"environments": [
		{
			"WindowsSDKVersion": "8.1"
		}
	], 
```
I expected to be able to do the same in vscode by setting WindowsSDKVersion in the environmentVariables of the Kit to force vcvarsall.bat to select the correct SDK. My kit looks like this:
```
  {
    "name": "VS2017_x64_SDK8.1",
    "visualStudio": "dd9cc0c1",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Ninja"
    },
    "environmentVariables": {
      "WindowsSDKVersion": "8.1"
    }
  },
```

However, it seems that currently the environment variables defined in environmentVariables for a Visual Studio kit are just appended to the list of variables generated by vcvarsall.bat. It would be very useful if these environment variables were set when calling vcvarsall.bat to control its setup. As of now, I don't see any other way of doing it than setting those variables before starting vscode...

**Comment by bobbrow** [comment]
I ran the vcvars script in a CMD window with WindowsSDKVersion preset to 8.1 and the vcvars script overwrote my setting. So it would seem you want the environment variables applied after the vcvars variables, no? Can you help me understand what the expectation is? We could potentially inject the requested environment into the process that runs the vcvars script, but I want to make sure that has the desired effect before we make a change here.  Thank you!

**Comment by Djoulihen** [comment]
Hi Bob, sorry for the late reply I was out-of-office last week.

I've made some more tests on my side today to better understand how the Windows SDK version can be controlled when running vcvars. As you said the script overrides any previously set value of WindowsSDKVersion. Now I truly don't understand how my settings work in Visual Studio 2019 since having WindowsSDKVersion=8.1 in my CMakeSettings.json correctly sets all environment variables to the 8.1 SDK as if I'd passed the correct option to vcvars This led me to (incorrectly) believe that setting the environment variable before calling the script should work. I have no idea why this works in Visual Studio as this seems to be a totally undocumented and maybe unintended "feature" which I just happened to find by trial and error. Maybe you guys have more insights into this than I do...

Returning to the problem at hand, after some digging in the scripts, I found another environment variable that I could set prior to calling vcvars to control the windows SDK version: __VSCMD_ARG_WINSDK. However this seems like in internal variable, subject to change, and feels kind of hackish to me... 

Being able to override the environment variables set by the script, as you mentioned, could be an idea but the SDK version is present in many variables and highly intertwined with the toolset version etc. So at this point I might as well get rid of the MSVC specific setup (using vcvars) and create the whole environment, replicating the result of calling vcvars with the correct option.

After looking at this in more details, maybe I'm missing something but I've come to the conclusion that the proper way of doing this would be to pass the windows SDK version as a parameter to vcvarsall.bat as it's intended to work that way. This could be an additional parameter of the Kit (like "visualStudioArchitecture") but controlling the SDK version that is passed to vcvarsall.bat as its 3rd (or 2nd if no platform type is specified) argument, maybe something like "visualStudioWinSDKVersion". As this is a very different issue from the one I initially opened, let me know if you'd like me to open a new one, add a comment to #125 or modify the title of this one.
